Реализация взаимодействия с БД

rvkhlystov opened the pull request:
Подключил БД H2, создал AccountEntity и ClientEntity. Поля не мапятся на таблицу. При запросе получаю null

rvkhlystov committed:
Подключил БД H2, создал AccountEntity и ClientEntity. Поля не мапятся на таблицу. При запросе получаю null

RodinDenis commented on server/src/main/java/ru/sbrf/payment/server/check/CheckClient.java:
```diff
@@ -1,6 +1,7 @@
+import ru.sbrf.payment.server.client.Client;
 
 import java.util.HashMap;
 import java.util.Optional;
```
таким образом использовать Optional для проверки на null не стоит. Если есть желание использовать Optional, то лучше изменить архитектуру таким образом чтобы из clients.get возвращался Optional

rvkhlystov committed:
Начал реализацию работы с БД. Привязал классы Entity к БД с клиентами. Реализовал временную логику,согласно которой находится клиент в БД, и далее проводится платеж.

Необходимо:
1. реализовать списание в БД
2. создать БД платежей и реализовать запись платежей в эту БД

rvkhlystov committed:
Реализовал:
1. слой взаимодействия с БД через сервис DBInteractionService.
2. списание суммы платежа с баланса
3. создал таблицу платежей

Необходимо:
1. реализовать запись платежей в таблицу платежей
2. избавиться от неправильной реализации Optional в классах Check

rvkhlystov committed:
Реализовал:
1. реализовал запись платежей в таблицу платежей

Необходимо:
1. избавиться от излишней логики
2. устранить NPE на третьей транзакции
3. избавиться от неправильной реализации Optional в классах Check
4. заменить "гвозди" в Currency на CreatorCurrency
